install/kubernetes/cilium/templates/cilium-nodeinit/daemonset.yaml

```diff
@@ -46,19 +46,6 @@ spec:
           image: {{ include "cilium.image" .Values.nodeinit.image | quote }}
           imagePullPolicy: {{ .Values.nodeinit.image.pullPolicy }}
           lifecycle:
-            {{- if .Values.eni.enabled }}
-            postStart:
-              exec:
-                command:
-                  - nsenter
-                  - --target=1
-                  - --mount
```

Comment:
So presumably is executing the script in the host mount namespace. I assume this is to use the hosts' iptables binary, to ensure we're using the same backend as AWS VPC CNI (i.e. iptables-nft vs iptables-legacy).

Are we sure that if if we now execute this inside the cilium-agent (which iirc uses iptables-wrapper) we're still using the correct backend? Might be worth checking what version of iptables the AWS VPC CNI ships to ensure we're not accidentally missing the chain because we selected the wrong backend.

Source is here: https://github.com/aws/amazon-vpc-cni-k8s

Reply:
It seems the backend is actually configurable :scream: https://github.com/aws/amazon-vpc-cni-k8s#enable_nftables-v1121

I'm not sure if the old code was actually correct about the choice of backend

Reply:
Another option would be to try to remove the chain in both backends maybe?

Reply:
That's a good point that I haven't considered. Will check.

Reply:
I did some digging. It seems the reason why we were mounting the host namespace is due to iptables requiring the `xtables.lock` that needs to be shared with the host https://github.com/cilium/cilium/pull/20741

So running it inside the Cililum agent container is fine, since there we do have access to the xtables lock.

I'm still wondering though if we should try to be smart about iptables backend selection - but that can be done in a follow-up PR, since the old code is likely also not handling it either.

Reply:
@gandro so the AWS VPC CNI uses `coreos/go-iptables` which uses `iptables` (the cni uses `New` without any options provided, so no `ip6tables`) command: https://github.com/coreos/go-iptables/blob/14d56d57c892f27a717aa6026fd2d3293221395b/iptables/iptables.go#L537

As far as I understand the iptables-wrapper docs: https://github.com/kubernetes-sigs/iptables-wrappers#iptables-wrappers , it should automatiocally figure out what backend is used on host and link to correct underlying binary.

Reply:
Thanks a lot for the research, yeah I think this is good to go!